app/upgrades/v2_5_0/v2_5_0.go

```diff
@@ -248,138 +144,110 @@ func UpgradeStNibiContractOnMainnet(
 		GasPrice:         unusedBigInt,
 		GasFeeCap:        unusedBigInt,
 		GasTipCap:        unusedBigInt,
-		Data:             input, // This manages the constructor args
+		Data:             contractInput, // This manages the constructor args
 		AccessList:       gethcore.AccessList{},
 		SkipNonceChecks:  false,
 		SkipFromEOACheck: false,
 	}
-	sdb = nibiru.EvmKeeper.NewStateDB(
-		ctx,
-		nibiru.EvmKeeper.TxConfig(ctx, gethcommon.Hash{}),
-	)
-	evmObj = nibiru.EvmKeeper.NewEVM(
-		ctx,
-		evmMsg,
-		nibiru.EvmKeeper.GetEVMConfig(ctx),
-		nil, // tracer: unused
-		sdb,
-	)
+	sdb := keepers.EvmKeeper.NewStateDB(ctx, keepers.EvmKeeper.TxConfig(ctx, gethcommon.Hash{}))
+	evmObj := keepers.EvmKeeper.NewEVM(ctx, evmMsg, keepers.EvmKeeper.GetEVMConfig(ctx), nil, sdb)
 
-	evmResp, err := nibiru.EvmKeeper.CallContractWithInput(
-		ctx, evmObj, evmMsg.From, nil, true /*commit*/, input,
+	evmResp, err := keepers.EvmKeeper.CallContractWithInput(
+		ctx, evmObj, evmMsg.From, nil, true /*commit*/, contractInput,
 		evmkeeper.Erc20GasLimitDeploy,
 	)
 	if err != nil {
 		return fmt.Errorf("failed to deploy ERC20 contract: %w", err)
 	} else if len(evmResp.VmError) > 0 {
 		return fmt.Errorf("VM Error in deploy ERC20: %s", evmResp.VmError)
 	}
-	evmResp.Logs = append(evmLogs, evmResp.Logs...)
+
+	evmLogs = append(evmLogs, evmResp.Logs...)
 	_ = ctx.EventManager().EmitTypedEvents(
 		&evm.EventContractDeployed{
 			Sender:       evmMsg.From.Hex(),
-			ContractAddr: erc20AddrForNewDeploment.Hex(),
+			ContractAddr: newErc20Addr.Hex(),
 		},
 	)
+	fmt.Println("deployed ERC20 contract at", newErc20Addr.Hex())
 
 	// -------------------------------------------------------------------------
-	// STEP 5: Sanity check the new contract with address "erc20AddrForNewDeploment"
```

Comment:
We don't need to check the metadata on the new ERC20 contract because we only care about it on the original contract, so I moved this check to the end of the upgrade handler.